conda_build/index.py

```diff
@@ -474,6 +421,9 @@ def _clear_newline_chars(record, field_name):
             record[field_name] = record[field_name][0].strip().replace("\n", " ")
 
 
+@deprecated(
+    "24.1.0", "24.5.0", addendum="Use `conda_index._apply_instructions` instead."
+)
 def _apply_instructions(subdir, repodata, instructions):
```

Comment:
I marked this explicitly as deprecated since I know there are some conda-forge scripts that use `_apply_instructions`.
(We probably shouldn't to continue to use the underscore-prefixed function from `conda-idex` either. But this way we can at least mark it for removal here.)

I assume we don't have to mark every single underscore-prefixed symbol here separately, right?!

Reply:
Every function that we know is used and plan to remove, independent whether it's prefixed with an underscore or not, is going to need the decorator.

Remember [Hyrum's Law](https://www.hyrumslaw.com/), everything is API in the eye of the user, we can only manage the way it's being used and informing the users about our intentions, is good engineering practice.

Reply:
> Every function that we know is used \[...\]

The problem is, we of course don't know which are used in the wild...
I've now just plonked the decorator on everything in `conda_build.index` :shrug:.

(If anyone knows a convenient way to really handle API<->internals definition in Python which we could use for future modules/packages, I'm all ears. Python, since it's one of those "everything is public and dynamic" languages, is just awful when Mr. Wright et al. point out these issues...)